### DISCUSSION TITLE
Computed fields -- model_dump and equals

### DISCUSSION THREAD

**Discussion by eg-novelt** [pydantic/pydantic#7146] [Pydantic V2]
Is it intended that computed fields are included in model_dump()?  

Also, I found that the presence of computed fields can alter if 2 models are considered ==; is that intended ?  Because computed fields should be 100% dependent on existing fields, this seems at first glance wrong.  Although, I suppose you can't rely on an implementation conforming to that rule.  For example, a computed field included the date it was computed.  

For my purposes, just including @cached_property and not @computed_field does what I want.

It's nice that pydantic didn't reinvent the wheel :)

**Comment by danielniccoli** [comment]
> Is it intended that computed fields are included in model_dump()?

Yes.

> For example, a computed field included the date it was computed.

A computed field is roughly equivalent to a pure Python [property getter](https://docs.python.org/3/library/functions.html?highlight=property#property). If your getter method returns a random number, it'll be a random number every time you get the value. If you want a computed value that is computed only on instantiation, use `default_factory`.

```python
class MyClass(BaseModel):
    _guid = PrivateAttr(default_factory=lambda: uuid4())
    
    @computed_field
    @property
    def guid(self) -> str:
        return self._guid
```

Computed fields behave exactly like you define them. If you return a random number in a computed field, then it'll be different each time you call the computed field.

> For my purposes, just including @cached_property and not @computed_field does what I want.

The docs for `@cached_property` don't explain, what exactly happens here. But my assumption based on the name of the decorator is that  computation does not happen every time you get the value, but it probably also does not ensure the value is *never* recomputed. If your computed property does not behave as you expect, using a cached property will likely cause bugs down the line.
